File: src/main/java/ru/citeck/idea/utils/MavenUtils.java
package ru.citeck.idea.utils;

import com.intellij.openapi.module.Module;
import com.intellij.openapi.module.ModuleUtil;
import com.intellij.openapi.project.ProjectUtil;
import com.intellij.openapi.vfs.VirtualFile;
import com.intellij.psi.PsiElement;
import com.intellij.psi.util.PsiUtil;
import com.intellij.psi.xml.XmlFile;

import java.util.Optional;

public class MavenUtils {

    public static XmlFile getPomFile(PsiElement psiElement) {
        Module module = ModuleUtil.findModuleForPsiElement(psiElement);
        return getPomFile(module);
    }

    public static XmlFile getPomFile(Module module) {

        if (module == null) {
            return null;
        }

        VirtualFile moduleDir = ProjectUtil.guessModuleDir(module);
        if (moduleDir == null) {
            return null;
        }

        return Optional
            .ofNullable(moduleDir.findChild("pom.xml"))
            .map(vFile -> (PsiUtil.getPsiFile(module.getProject(), vFile)))
            .filter(XmlFile.class::isInstance)
            .map(XmlFile.class::cast)
            .orElse(null);
    }
}

File: src/main/kotlin/ru/citeck/idea/utils/MavenUtils.kt
package ru.citeck.idea.utils

import com.intellij.openapi.module.Module
import com.intellij.openapi.module.ModuleUtil
import com.intellij.openapi.project.guessModuleDir
import com.intellij.openapi.vfs.VirtualFile
import com.intellij.psi.PsiElement
import com.intellij.psi.PsiFile
import com.intellij.psi.util.PsiUtil
import com.intellij.psi.xml.XmlFile
import java.util.*

object MavenUtils {

    fun getPomFile(psiElement: PsiElement): XmlFile? {
        val module = ModuleUtil.findModuleForPsiElement(psiElement)
        return getPomFile(module)
    }

    fun getPomFile(module: Module?): XmlFile? {

        if (module == null) {
            return null
        }

        val moduleDir = module.guessModuleDir() ?: return null

        return Optional
            .ofNullable(moduleDir.findChild("pom.xml"))
            .map { vFile: VirtualFile? ->
                (PsiUtil.getPsiFile(
                    module.project,
                    vFile!!
                ))
            }
            .filter { obj: PsiFile? -> XmlFile::class.java.isInstance(obj) }
            .map { obj: PsiFile? -> XmlFile::class.java.cast(obj) }
            .orElse(null)
    }
}
